tests/test_jobs.py

```diff
@@ -89,6 +89,34 @@ def test_job_start_blocking(default_domino_client):
     job = default_domino_client.job_start_blocking(command="main.py")
     assert job["statuses"]["executionStatus"] == "Succeeded"
 
+@pytest.mark.skipif(not domino_is_reachable(), reason="No access to a live Domino deployment")
+def test_job_start_override_hardware_tier_id(default_domino_client):
+    """
+    Confirm that we can start a job using the v4 API and override the hardware tier id
+    """
+    hardware_tiers = default_domino_client.hardware_tiers_list()
+    non_default_hardware_tiers = [hwt for hwt in hardware_tiers if not hwt["hardwareTier"]["isDefault"]]
+    if len(list(non_default_hardware_tiers)) > 0:
+        override_hardware_tier_id = non_default_hardware_tiers[0]["hardwareTier"]["id"]
+        job_status = default_domino_client.job_start_blocking(command="main.py", hardware_tier_id=override_hardware_tier_id)
+        assert job_status['statuses']['isCompleted'] is True
+        job_red = default_domino_client.job_runtime_execution_details(job_status['id'])
+        assert job_red["hardwareTierId"] == override_hardware_tier_id
+
+# deprecated but ensuring it still works for now
+@pytest.mark.skipif(not domino_is_reachable(), reason="No access to a live Domino deployment")
+def test_job_start_override_hardware_tier_name(default_domino_client):
+    """
+    Confirm that we can start a job using the v4 API and override the hardware tier via hardware_tier_name
+    """
+    hardware_tiers = default_domino_client.hardware_tiers_list()
+    non_default_hardware_tiers = [hwt for hwt in hardware_tiers if not hwt["hardwareTier"]["isDefault"]]
+    if len(list(non_default_hardware_tiers)) > 0:
+        override_hardware_tier_name = non_default_hardware_tiers[0]["hardwareTier"]["name"]
+        job_status = default_domino_client.job_start_blocking(command="main.py", hardware_tier_name=override_hardware_tier_name)
+        assert job_status['statuses']['isCompleted'] is True
+        job_red = default_domino_client.job_runtime_execution_details(job_status['id'])
+        assert job_red["hardwareTier"] == override_hardware_tier_name
```

Comment:
The way this is written, it leaves open the possibility of a falsely passing test. If `len(non_default_hardware_tiers) == 0`, then the actual test never runs -- but it will still appear to be "passing." Reverse the logic and `xfail()` if the test can't run.

```suggestion
    if len(list(non_default_hardware_tiers)) > 0:
        pytest.xfail("No non-default hardware tiers found: cannot run test")

    override_hardware_tier_name = non_default_hardware_tiers[0]["hardwareTier"]["name"]
    job_status = default_domino_client.job_start_blocking(command="main.py",
                                                          hardware_tier_name=override_hardware_tier_name)
    assert job_status['statuses']['isCompleted'] is True
    job_red = default_domino_client.job_runtime_execution_details(job_status['id'])
    assert job_red["hardwareTier"] == override_hardware_tier_name
```